### PR TITLE
Fix duplicate test code in helm history command

### LINKED ISSUE #3649: Possible duplication in history command tests
It seems there is some duplication of test code under cmd/helm/history_test.go:

https://github.com/kubernetes/helm/blob/abe958eb872f41a6237fcb8f3c65f2aa96763c52/cmd/helm/history_test.go#L38-L42

and 

https://github.com/kubernetes/helm/blob/abe958eb872f41a6237fcb8f3c65f2aa96763c52/cmd/helm/history_test.go#L68-L82
This can use the `releaseCase` structure and the corresponding `runReleaseCases` function for all the tests.

### PR DESCRIPTION
**Pull request by adshmh**
fix(helm): remove duplicate code from cmd/helm/history_test.go. Closes #3649.

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

### CODE REVIEW

File: cmd/helm/history_test.go [hunk 1]
```diff
@@ -34,50 +35,31 @@ func TestHistoryCmd(t *testing.T) {
...
-		desc string
-		args []string
-		resp []*rpb.Release
-		xout string
-	}{
+	tests := []releaseCase{
 		{
-			cmds: "helm history RELEASE_NAME",
-			desc: "get history for release",
+			name: "helm history RELEASE_NAME -- get history for release",
```
Comment (thomastaylor312) [inline cmd/helm/history_test.go]: I think on these names, let's just shorten to something like `get history for release` and drop the rest

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
One tiny change request here to make the test output more clear. Thanks again for another great refactoring PR!

File: cmd/helm/history_test.go (hunk 1 above)
Comment (adshmh) [reply cmd/helm/history_test.go]: Thank you for the review. I have updated the PR using shortened names for the tests.

**Review by thomastaylor312 — APPROVED** [review]
Thank you for the fix!

**Pull request merged**